src/modules/kubernetes.rs

```diff
@@ -0,0 +1,54 @@
+use ansi_term::Color;
+use std::process::Command;
+
+use super::{Context, Module};
+
+fn ex(cmd: &str, args: &[&str]) -> Result<String, String> {
+    let output = Command::new(cmd)
+        .args(args)
+        .output()
+        .map_err(|e| e.to_string())?;
+    let output = String::from_utf8(output.stdout).map_err(|e| e.to_string())?;
+    Ok(output)
+}
+
+/// Creates a module with the current k8s context/namespace
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let maybe_ctx: Option<String> = ex("kubectl", &["config", "current-context"])
+        .map(Some)
+        .or::<String>(Ok(None))
+        .unwrap();
+    let maybe_namespace: Option<String> = ex(
```

Comment:
What's the runtime cost of running `kubectl`, especially if your caches are empty?

Most of our modules are gated behind the existence of files, which is a relatively quick check, then a more expensive call to runtimes if the appropriate files are found. Without that, we'd be forking a process for each module every time we ran the shell, which isn't exactly *slow* individually, but adds up.

Is there the presence of some sort of file/envar or other quick check we could use as a fast check before running kubectl?

Reply:
@chipbuster Possibly.  I can look into checking for `$KUBECONFIG` (iirc) and then parsing that file.  I think all of the data is in there somewhere.